### PR TITLE
Publish to gateway(edit velocity temp) - only skipGateway disabled apps

### PR DESCRIPTION
**Pull request by lahiru823**

### CODE REVIEW

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java [hunk 1]
```diff
@@ -617,7 +617,11 @@ public void updateAPI(WebApp api) throws AppManagementException {
                             WebApp apiPublished = getAPI(api.getId());
                             apiPublished.setOldInSequence(oldApi.getInSequence());
                             apiPublished.setOldOutSequence(oldApi.getOutSequence());
-                            publishToGateway(apiPublished);
+
+                            //publish to gateway if skipGateway is disabled only
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:621]: Start comments with a capital letter and end with a full stop.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java [hunk 2]
```diff
@@ -806,7 +810,11 @@ public void changeAPIStatus(WebApp api, APIStatus status, String userId,
                 if (gatewayType.equalsIgnoreCase(AppMConstants.API_GATEWAY_TYPE_SYNAPSE) && updateGatewayConfig) {
                     if (status.equals(APIStatus.PUBLISHED) || status.equals(APIStatus.DEPRECATED) ||
                         status.equals(APIStatus.BLOCKED)) {
-                        publishToGateway(api);
+
+                        //publish to gateway if skipGateway is disabled only
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:814]: Start comments with a capital letter and end with a full stop.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 3]
```diff
@@ -4169,11 +4146,18 @@ public WorkflowDTO retrieveLatestWorkflowByReference(String workflowReference) t
...
         WorkflowDTO workflowDTO = null;
+        String query;
 
-        String query = "SELECT * FROM APM_WORKFLOWS WHERE WF_REFERENCE=? ORDER BY WF_CREATED_TIME LIMIT 1";
         try {
-
             connection = APIMgtDBUtil.getConnection();
+
+            //oracle specific query
+            if (connection.getMetaData().getDriverName().contains("Oracle")) {
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4155]: Make the database name a constant.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 4]
```diff
@@ -7268,21 +7276,23 @@ public static NativeArray getAvailableJavaPolicyList(String applicationUUId, boo
 		NativeObject objPolicy;
 		NativeArray arrJavaPolicies = new NativeArray(0);
 		Integer count = 0;
+        Boolean isMandatory = false; //no need to show the mandatory fields as options
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7279]: Formatting issues.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 5]
```diff
@@ -7268,21 +7276,23 @@ public static NativeArray getAvailableJavaPolicyList(String applicationUUId, boo
...
 				"ORDER BY DISPLAY_ORDER_SEQ_NO  ";
 
 		try {
 			conn = APIMgtDBUtil.getConnection();
 			ps = conn.prepareStatement(query);
 			ps.setString(1, applicationUUId);
-			ps.setBoolean(2, isGlobalPolicy);
-			rs = ps.executeQuery();
+			ps.setBoolean(2, isMandatory);
+            ps.setBoolean(3, isGlobalPolicy);
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7294]: Formatting issues.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 6]
```diff
@@ -7361,21 +7371,23 @@ public void saveJavaPolicyMappings(Connection connection, int applicationId, Obj
 		ResultSet rs = null;
 		List<JavaPolicy> policies = new ArrayList<JavaPolicy>();
 		String strJavaPolicyProperty = "";
+        boolean isMandatory = true;
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7374]: Formatting issues.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java [hunk 7]
```diff
@@ -7361,21 +7371,23 @@ public void saveJavaPolicyMappings(Connection connection, int applicationId, Obj
...
 
 		try {
 			conn = APIMgtDBUtil.getConnection();
 			ps = conn.prepareStatement(query);
 			ps.setString(1, applicationUUId);
-			ps.setBoolean(2, isGlobalPolicy);
-			rs = ps.executeQuery();
+			ps.setBoolean(2, isMandatory);
+            ps.setBoolean(3, isGlobalPolicy);
+            rs = ps.executeQuery();
```
Comment (rushmin) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7390]: Formatting issues.

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java (hunk 1 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:621]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/67fe5d39def098a363813085ad1c01d72ec5b2d0

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java (hunk 2 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/APIProviderImpl.java:814]: fixed with commit# https://github.com/wso2/carbon-appmgt/commit/67fe5d39def098a363813085ad1c01d72ec5b2d0

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 3 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:4155]: commit# https://github.com/wso2/carbon-appmgt/commit/da3135c15509f5a124531f4d286f9c017086e98f

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 4 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7279]: Already fixed with a subsequent fix

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 5 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7294]: Already fixed with a subsequent fix

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 6 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7374]: Already fixed with a subsequent fix

File: components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java (hunk 7 above)
Comment (lahiru823) [inline components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java:7390]: Already fixed with a subsequent fix

**Pull request merged**